### ISSUE TITLE
Defining symbolic attributes not in initialization

### ISSUE THREAD

**Issue by sscardapane** [google/pyglove#116]
I apologize in advance if this is a stupid question, but I really like the design of the library and I am afraid I am missing some context to "grok" it. Consider a simple MLP in PyTorch:

```@pg.symbolize
class SimpleMLP(Module):

  def __init__(self, in_channels, hidden_channels, out_channels, activation=F.relu):
    super().__init__()
    self.hidden = Linear(in_channels, hidden_channels)
    self.out = Linear(hidden_channels, out_channels)
    self.activation = activation

  def forward(self, x):
    return self.out(self.activation(self.hidden(x)))
```

Assume I now want to patch the linear layers to always add dropout afterwards. Is there a way to manipulate the decorator to force this? All the examples in the documentation use variants of `Sequential`, where the layers are explicitly passed during initialization. Since this is a rather common use case I am wondering if I completely misunderstanding the documentation and there is an easy way to accomplish this, or else if I should refactor the above code. Thanks!

**Comment by daiyip** [comment]
Have you tried `pg.detour`? Consider the following:

```python
def with_dropout(cls, *args, **kwags):
  return Sequential([
     cls(*args, **kwargs),
     Dropout(0.2)
  ])

# Under the context manager, all instantiation of `Linear` will be detoured to `with_dropout` function.
# You can insert the context manager at the right place to decide what to replace and what not to.
with pg.detoru([(Linear, with_dropout)]):
  mlp = SimpleMLP(48, 32, 128)
  mlp.forward(input)
```

See [the doc](https://pyglove.readthedocs.io/en/latest/learn/soop/detour.html) of`pg.detour` for more details.

Let me know if this works for you.

**Comment by sscardapane** [comment]
This is an interesting solution, thanks!
